### DISCUSSION TITLE
Event colors does not seem to reset to default calendar color

### DISCUSSION THREAD

**Discussion by blip-unicorn** [bitfireAT/ical4android#177] [General]
I use colors to mark different states of events. When all is prepared and booked I reset the color to the default, this change does not make it all the way to the calendar app on my Android phone.

**My setup is:**
Nextcloud 29.0.8
Etar (did not find version)
Davx5 4.4.3.2-ose

**Steps to reproduce:**
Create an calendar event in Nextcloud Calendar, select a color for the event.
Sync to Etar, notice that the color is set correctly.
In Nextcloud press the curvy arrow to restore color to the default, change the time of the event.
Sync again.

**Actual result:**
In Etar, the time is updated but the color stays the same as before.

**Expected result:**
Time and color of the event is updated.

**Known workaround:**
Do not use default color, there is no problem when changing from one color to another.

**Additional information;**
I have looked in the Nextcloud database and the .ics returned from the CalDav call. When the color is reset the COLOR property of the iCalendar object is removed. My Kotlin is not good enough to see how Dav5x handles the event syncing in detail, but my guess is that extra work needs to be done as the property is removed by the server.

Is this a bug or a known limitation of how Android calendars work?

**Comment by rfc2822** [comment]
I can confirm that the color is persists, although `AndroidEvent.update()` sets `EVENT_COLOR_KEY=null` when the updated event doesn't have a COLOR property.

~~~
ContentProviderOperation(type=update uri=content://com.android.calendar/events/5?account_name=test%40example.com&account_type=bitfire.at.davdroid&caller_is_syncadapter=true values={organizer=test@example.com, duration=null, sync_data1=8d0949e68cd4372381f95860a4b04497, sync_data2=1, sync_data3=5, sync_data4=null, description=null, allDay=0, exdate=null, exrule=null, accessLevel=3, hasAttendeeData=0, uid2445=ef17aea0-7927-4043-83ef-a1634394eded, eventLocation=null, dirty=0, dtend=1730196000000, eventStatus=1, rdate=null, rrule=null, title=Kein-Farbtermin, eventEndTimezone=Europe/Vienna, eventTimezone=Europe/Vienna, calendar_id=1, _sync_id=186F63F8-C369-41AA-8454-F360B04FDEEE.ics,
--> eventColor_index=null, <--
deleted=0, dtstart=1730192400000, availability=0})
~~~

**Reply by rfc2822** [reply]
See #178

**Comment by blip-unicorn** [comment]
Thank you for an prompt and informative answer!

**Comment by blip-unicorn** [comment]
With version 4.4.4-ose this issue has been resolved.
